File: 1302-delete-characters-to-make-fancy-string/delete-characters-to-make-fancy-string.cpp
class Solution {
public:
    string makeFancyString(string s) {
        if(s.size()<3) return s;
        int i = 2, j = 2;
        for(int i=2;i<s.size();i++) {
            if(s[i]!=s[j-1] || s[i]!=s[j-2]) {
                s[j] = s[i];
                j++;
            }
        }
        s.resize(j);
        return s;
    }
};

File: 1302-delete-characters-to-make-fancy-string/delete-characters-to-make-fancy-string.java
class Solution {
    public String makeFancyString(String s) {
        if(s.length()<3) return s;
        StringBuilder sb = new StringBuilder(s);
        int j = 2;
        
        for (int i = 2; i < s.length(); i++) {
            if (s.charAt(i) != sb.charAt(j - 1) || s.charAt(i) != sb.charAt(j - 2)) {
                sb.setCharAt(j, s.charAt(i));
                j++;
            }
        }
        
        return sb.substring(0, j);
    }
}